util/reggen/field.py

```diff
@@ -116,6 +125,18 @@ def from_raw(reg_name: str,
         else:
             hwaccess = default_hwaccess
 
+        raw_hwqe = rd.get('hwqe')
```

Comment:
Probably better: `raw_hwqe = rd.get('hwqe', default_hwqe)`, then you don't have to handle the `None` case below.

Reply:
thanks @rswarbrick , i always forget you can put a default value there...

Reply:
The test for None was on purpose here, though, so I can see if it was specified at the field level. Perhaps I can rework the check for hwext and illegal values--I just wanted to avoid writing the same exception code in two places.

Reply:
If you call the `get` method on a dictionary, you can pass a default value to use when the key doesn't exist. My suggestion does the same as your code, but less verbosely. If you look around the rest of the reggen code, you'll see the same trick elsewhere.